langchain/document_loaders/audio.py

```diff
@@ -0,0 +1,35 @@
+import os
+import openai
+from typing import List
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+class AudioFileLoader(BaseLoader):
```

Comment:
@rlancemartin :wave: 

This audio file loader is coupled to the file system right now, and allows loading a single file at a time. 

This is OK, but users may find it a bit more convenient to be able to scan an entire directory of audio files and decide which kinds of file to pick up (e.g., mp3s, wav) etc, get tqdm progress bar, and eventually get both sync and async implementations that are concurrent (the concurrency part is not implemented yet).

Let's replace: 

* Instead of inheriting from `BaseLoader` let's inherit from `GenericLoader` 
 https://github.com/hwchase17/langchain/blob/5aa7264a63e13ed550b7c85cbabdf0afcdab390b/langchain/document_loaders/generic.py#L17-L17

The baseclass will introduce convenience classmethods that will allow picking up audio files from the file system, add progress bars, etc.

* Let's introduce a `BlobParser` for Audio (say `OpenAIWhisperParser` or something like that with a good name). It should live in `document_loaders.parsers.audio.py` (can follow the structure of any other blob parsers)

--- 

Testing: 

Ideally, we can add a unit-test, we should be able to patch the request from openai.Audio.transcribe to return a mock transcription.

Reply:
Thanks for the review! 

> Instead of inheriting from BaseLoader let's inherit from GenericLoader

Done. I see `GenericLoader` uses  `FileSystemBlobLoader`, which I saw from your [doc](https://www.notion.so/LangChain-Documents-786e0b2762694ca786889556ea4586ba) gives us some useful [methods](https://github.com/hwchase17/langchain/blob/c4b502a47051f50c6e24b824d3db622748458d13/langchain/document_loaders/blob_loaders/file_system.py#L39-L39) (e.g., progress bar) as you mention.